IPython/terminal/embed.py

```diff
@@ -83,6 +96,8 @@ def __init__(self, **kw):
 
         super(InteractiveShellEmbed,self).__init__(**kw)
 
+        frame = sys._getframe(InteractiveShellEmbed._stack_frames)
+        self._call_location_id = '%s:%s' % (frame.f_code.co_filename, frame.f_lineno)
```

Comment:
Oops, sorry for not reading that.

Why not have `embed()` get its caller from the stack and pass it in to `InteractiveShellEmbed()` as an optional parameter. Then `InteractiveShellEmbed()` can do the right thing if it's not passed in.